scenario/clickactionbutton.go

```diff
@@ -0,0 +1,643 @@
+
+	switch t := obj.EnigmaObject.(type) {
+	case *enigma.GenericObject:
+		genericObj := obj.EnigmaObject.(*enigma.GenericObject)
+
+		propsRaw, err := sessionState.SendRequestRaw(actionState, genericObj.GetPropertiesRaw)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "Properties request failed for object<%s>", obj.ID)
+		}
+
+		// parse button actions
+		actionsRaw, err := senseobjdef.NewDataPath("actions").Lookup(propsRaw)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, `No "actions"-property exist for object<%s>`, obj.ID)
+		}
+
+		actions := []buttonAction{}
+		err = json.Unmarshal(actionsRaw, &actions)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "Failed to unmarshal button actions")
+		}
+
+		// parse navigation action associated with button
+		navigationRaw, err := senseobjdef.NewDataPath("navigation").Lookup(propsRaw)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, `No "navigation"-property exist for object<%s>`, obj.ID)
+		}
+
+		navigationAction := &buttonNavigationAction{}
+		err = json.Unmarshal(navigationRaw, navigationAction)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "Failed to unmarshal button navigation action")
+		}
+
+		return actions, navigationAction, nil
+
+	default:
+		return nil, nil, errors.Errorf("Expected generic object , got object type<%T>", t)
+	}
+}
+
+// execute one action contained in a Sense action-button
+func (buttonAction *buttonAction) execute(sessionState *session.State, actionState *action.State) error {
+	uplink := sessionState.Connection.Sense()
+	doc := DocWrapper{uplink.CurrentApp.Doc}
+	sendReq := func(f func(context.Context) error) error {
+		return sessionState.SendRequest(actionState, f)
+	}
+
+	switch buttonAction.ActionType {
+	case emptyAction: // do nothing
+		return nil
+
+	case unknownAction:
+		return errors.New("Unknown button action")
+
+	case applyBookmark:
+		return sendReq(func(ctx context.Context) error {
+			success, err := doc.ApplyBookmark(ctx, buttonAction.Bookmark /*id*/)
+			if err != nil {
+				return errors.Wrapf(err, "Error applying bookmark<%s>", buttonAction.Bookmark)
+			}
+			if !success {
+				return errors.Errorf("Unsuccessful application bookmark<%s>", buttonAction.Bookmark)
+			}
+			return err
+		})
+
+	case moveBackwardsInSelections:
+		return sendReq(doc.Back)
+
+	case moveForwardsInSelections:
+		return sendReq(doc.Forward)
+
+	case clearAllSelections:
+		return sendReq(func(ctx context.Context) error {
+			return doc.ClearAll(ctx, buttonAction.SoftLock /*lockedAlso*/, "" /*stateName*/)
+		})
+
+	case clearSelectionsInOtherFields:
+		return sendReq(func(ctx context.Context) error {
+			field, err := uplink.FieldCache.LookupWithFallback(buttonAction.Field,
+				func(fieldName string) (*enigma.Field, error) {
+					return doc.GetField(ctx, fieldName)
+				},
+			)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			_, err = field.ClearAllButThis(ctx, buttonAction.SoftLock)
```

Comment:
That's interesting! Maybe that would be a useful option to add to the ones in the Select action.